msbuild/custom-task-code-generation/README.md

```diff
@@ -0,0 +1,370 @@
+The task output will be:
+
+```csharp
+//The filename where the class was generated
+[Output]
+public string ClassNameFile { get; set; }
+```
+
+We need to override the Execute method. The execute method returns true if the task was successful and false in other cases. Task implements ITask and provides default implementations of some ITask members and additionally, logging is easier. It is important the log to know what is going on. And even more important if we are going to return not succeed (false). On error, we should use Log.LogError.
+
+```csharp
+public override bool Execute()
+{
+    //Read the input files and return a IDictionary<string, object> with the properties to be created.
+    //Any format error it will return not succeed and Log.LogError properly
+    var (success, settings) = ReadProjectSettingFiles();
+    if (!success)
+    {
+        return !Log.HasLoggedErrors;
+    }
+    //Create the class based on the Dictionary
+    success = CreateSettingClass(settings);
+
+    return !Log.HasLoggedErrors;
+}
+```
+
+Then, the details are really not important for our purpose. You can copy from the source code and improve if you like.
+
+:shipit:Food for thought. We are generating C# code during build process as example.The task is like any other C# class, you could do whatever you want. For example, sending an email, generating change log, reading github repository. This is the power of MSBuild custom tasks.
+
+### Step 3, Change the AppSettingStronglyTyped.csproj
+
+We need to make some changes on the project file. Now we have something simple like
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="17.0.0" />
+  </ItemGroup>
+
+</Project>
+```
+
+We are going to generate a NuGet package, so first we need to add some basic information
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+        <version>1.0.0</version>
+        <title>AppSettingStronglyTyped</title>
+        <authors>John Doe</authors>
+        <description>Generates a strongly typed setting class base on a txt file</description>
+        <tags>MyTags</tags>
+        <copyright>Copyright ©Microsoft Company 2022</copyright>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="17.0.0" />
+  </ItemGroup>
+
+</Project>
+```
+
+Then, the dependencies of your MSBuild task must be packaged inside the package, they cannot be expressed as normal PackageReferences. We don't expose any regular dependencies to the outside world. It is not needed for the current example, because we don't have extra dependencies, but it is worth being prepared for and be aware of.
```

Comment:
```suggestion
Then, the dependencies of your MSBuild task must be packaged inside the package, they cannot be expressed as normal PackageReferences. We don't expose any regular dependencies to the outside world. It is not needed for the current example, because we don't have extra dependencies, but it is worth being aware of this for other scenarios.
```